src/platform/Linux/GlibTypeDeleter.h

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <gio/gio.h>
+
+template<typename T, typename Deleter>
+class UniquePointerReceiver
+{
+public:
+    UniquePointerReceiver(std::unique_ptr<T, Deleter> & target) : mTarget(target) {}
+
+    ~UniquePointerReceiver() {
+        mTarget.reset(mValue);
+    }
+
+    T*& Get() {
```

Comment:
Why not `T** Get()` ?

This is only useful with C APIs, which use T** for pointer out params, right?